site/topics/objects/interacting-objects.rst

```diff
@@ -0,0 +1,63 @@
+***********************************
+Objects III --- Interacting Objects
+***********************************
+
+* We completed the ``Sphere`` class in the previous topic, and it will work for our needs
+* However, there are a few things we could do to improve the design and implementation
+* One idea is to make a ``Point3D`` class to manage all the point related things
+
+    * Location in three dimensional space
+    * Distance between points
+    * Equality on points
+
+* Although there is nothing wrong with the original ``Sphere`` implementation, let's consider that
+
+    * If we wanted to make other three dimensional shapes, they could use something like a ``Point3D`` class to represent their position
+    * It allows us to abstract and delegate some functionality to another class
+    * It will help with testing since we can test the ``Point3D`` class' functionality and the ``Sphere``\s independently
```

Comment:
It also makes `Sphere` smaller and easier to understand. You get to define methods based on  further abstractions, so there's less to think about. Checking if sphere's overlap, for example, can be checking if the distance between their centers is less than the sum of the radii -- ok, write that in code, no cruft about _how_ to calculate distance, the function you call worries about those details.

It can also allow for easier refactoring in the future. For example, you could change sphere to hypersphere by replacing `Point3D` with a generic n-dimensional `Point`. Then `Hypersphere` just works.

Reply:
> It can also allow for easier refactoring in the future. For example, you could change sphere to hypersphere by replacing `Point3D` with a generic n-dimensional `Point`. Then `Hypersphere` just works.

I love this point. I wont add it to the intro bit of the topic, but will hit it somewhere within the refactoring of ``Sphere`` in the latter part of the topic page.